### PR TITLE
feat: Alive interface for reporting Liveliness

### PR DESCRIPTION
**Pull request by fishstamp82**
This PR makes it possible to add custom liveliness handlers to restart containers on abnormal working conditions.

### CODE REVIEW

**Comment by drPytho** [comment]
I think this looks nice! Would love it if you added some documentation with it to let people know how they can use it :)

**Comment by drPytho** [comment]
A small test that just verifies that the code functions as expected too :)

**Review by drPytho — DISMISSED** [review]

**Review by drPytho — APPROVED** [review]

File: coverage.txt [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-github.com/voi-oss/svc/config.go:14.55,16.3 1 0
-github.com/voi-oss/svc/http.go:21.87,31.2 2 0
-github.com/voi-oss/svc/http.go:34.53,38.2 2 0
-github.com/voi-oss/svc/http.go:41.38,43.2 1 0
-github.com/voi-oss/svc/http.go:46.34,48.71 2 0
-github.com/voi-oss/svc/http.go:51.2,51.12 1 0
-github.com/voi-oss/svc/http.go:48.71,50.3 1 0
-github.com/voi-oss/svc/http.go:55.40,57.2 1 0
-github.com/voi-oss/svc/logger.go:13.102,28.2 5 19
+github.com/voi-oss/svc/config.go:11.44,13.2 1 2
```
Comment (K-Phoen) [inline coverage.txt:2]: Why is this file committed? :o

File: options.go [hunk 2]
```diff
@@ -125,8 +125,29 @@ func WithHealthz() Option {
...
+				return
+			}
+
+			s.logger.Warn("liveliness probe failed", zap.Errors("errors", errs))
+			b, err := json.Marshal(map[string]interface{}{"errors": errs})
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			w.WriteHeader(http.StatusServiceUnavailable)
```
Comment (K-Phoen) [inline options.go:148]: This starts the write process. Meaning that the headers set on the next line will be ignored. The order of these two lines should be inverted.
See https://pkg.go.dev/net/http#ResponseWriter

File: options_test.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
+package svc
+
+import (
+	"fmt"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
```
Comment (K-Phoen) [inline options_test.go:9]: `gofmt` ;)

**Review by K-Phoen — CHANGES_REQUESTED** [review]
Looking good, just a few comments that need addressing :)

File: coverage.txt (hunk 1 above)
Comment (fishstamp82) [reply coverage.txt:2]: It was modified when running `make test` locally.

File: options.go (hunk 2 above)
Comment (fishstamp82) [reply options.go:148]: Good catch, I am not sure if a specific content type is warranted here, its mainly the status code that is relevant (most of the time), but I will just flip the two lines.

Comment (K-Phoen) [reply options.go]: Yeah, agreed. I just commented because the content type was there :D

**Review by K-Phoen — APPROVED** [review]

**Pull request merged**